### ISSUE TITLE
[Bug] Schema setup & update jobs do not work for MySQL

### ISSUE THREAD

**Issue by fivos** [temporalio/helm-charts#310]
### What are you really trying to do?

Use the schema setup & update jobs to initialize the temporal & temporal_visibility schemas. 

### Describe the bug
Enabling either schema setup or update jobs fails with an error indicating that the job cannot connect to the SQL database. 

```bash
2022-08-05T23:19:28.104Z	ERROR	Unable to connect to SQL database.	{"error": "dial tcp 127.0.0.1:3306: connect: connection refused", "logging-call-at": "handler.go:54"}
```

<img width="1657" alt="Screen Shot 2022-08-05 at 4 35 16 PM" src="https://user-images.githubusercontent.com/294907/183223655-9c5f2fd0-a59d-42e4-b5ed-976b9c5f3225.png">

### Minimal Reproduction

1. Update values/values.mysql.yaml with the appropriate credentials to connect to a SQL database (username, password, etc)
2. Run
    ```sh
    helm install -f values/values.mysql.yaml temporaltest \
      --set schema.setup.enabled=true \
      --set schema.update.enabled=true . --timeout 900s
    ```

### Additional context
The root cause of the issue is that the SQL configuration parameters are not passed as environment variables. As described in [install-with-your-own-mysql](https://github.com/temporalio/helm-charts#install-with-your-own-mysql) we need environment variables such as SQL_USER, SQL_PASSWORD, SQL_HOST to be specified. 

We can see that in the generated YAML the list of environment variables is empty
```yaml
apiVersion: batch/v1
kind: Job
metadata:
  name: temporaltest-schema-setup
  labels:
    app.kubernetes.io/name: temporal
    helm.sh/chart: temporal-0.17.2
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/instance: temporaltest
    app.kubernetes.io/version: 1.17.2
    app.kubernetes.io/component: database
    app.kubernetes.io/part-of: temporal
  annotations:
    "helm.sh/hook": pre-install
    "helm.sh/hook-weight": "0"
    "helm.sh/hook-delete-policy": hook-succeeded,hook-failed
spec:
  backoffLimit: 100
  template:
    metadata:
      name: temporaltest-schema-setup
      labels:
        app.kubernetes.io/name: temporal
        helm.sh/chart: temporal-0.17.2
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/instance: temporaltest
        app.kubernetes.io/version: 1.17.2
        app.kubernetes.io/component: database
        app.kubernetes.io/part-of: temporal
    spec:
      
      restartPolicy: "OnFailure"
      initContainers:
          []
      containers:
        - name: default-schema
          image: "temporalio/admin-tools:1.17.2"
          imagePullPolicy: IfNotPresent
          command: ["temporal-sql-tool", "setup-schema", "-v", "0.0"]
          env:
        - name: visibility-schema
          image: "temporalio/admin-tools:1.17.2"
          imagePullPolicy: IfNotPresent
          command: ["temporal-sql-tool", "setup-schema", "-v", "0.0"]
          env:

```

**Comment by maordavidov** [comment]
Any news about this one? it almost a year open and unless temporal team expect no one to use their product it can be really helpful if you guys will fix it.

**Comment by mo-dayyeh** [comment]
I'm facing the issue with cassandra , I managed to fix it some how but it got broken again ! not sure why ! I'm using argocd application and the job never triggers even after cassandra DB is created :/

**Comment by robholland** [comment]
Fixed via #405
